packages/relay/src/RelayServer.ts

```diff
@@ -688,12 +692,32 @@ latestBlock timestamp   | ${latestBlock.timestamp}
     await this.withdrawToOwnerIfNeeded(block.number, block.hash, currentBlockTimestamp)
     this.lastRefreshBlock = block.number
     await this._refreshGasFees()
-    await this.registrationManager.refreshBalance()
-    if (!this.registrationManager.balanceRequired.isSatisfied) {
-      this.setReadyState(false)
+    const managerBalanceNotReady = await this._refreshAndCheckBalances()
+    if (managerBalanceNotReady) {
       return []
     }
-    return await this._handleChanges(block, currentBlockTimestamp)
+    return await this._handleChanges(block)
+  }
+
+  async _refreshAndCheckBalances (): Promise<boolean> {
+    const minBalanceToNotReadyFactor = 2
```

Comment:
Arbitrary hard-coded constant.

Also, why would server that is registered and staked be 'not ready' if its manager ran out of ether, if it has a worker that is fully funded? It can still mine a lot of transactions before there is any issue, right?